styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -247,6 +255,21 @@ private static boolean isActiveState(RunStateData runStateData) {
     final Schedule schedule;
 
     final WorkflowId workflowId = WorkflowId.create(input.component(), input.workflow());
+
+    final Workflow workflow;
+    try {
+      final Optional<Workflow> workflowResult = storage.workflow(workflowId);
+      if (workflowResult.isPresent()) {
+        workflow = workflowResult.get();
+      } else {
+        return Response.forStatus(
+            BAD_REQUEST.withReasonPhrase("The specified workflow is not found in the scheduler"));
+      }
+    } catch (IOException e) {
+      return Response.forStatus(
```

Comment:
Is this the same as throwing out `RuntimeException` with the message?

Reply:
No a `RuntimeException` would result in `500 Internal Server Error`. Don't want that.